pkg/daemon/server/service/runtime/pod_tracker.go

```diff
@@ -119,14 +120,15 @@ func (pt *PodTracker) addActivePod(vertexName string, index int) {
 	pt.activePodsMutex.Lock()
 	defer pt.activePodsMutex.Unlock()
 
-	pt.activePods[vertexName] = append(pt.activePods[vertexName], index)
+	if !slices.Contains(pt.activePods[vertexName], index) {
+		pt.activePods[vertexName] = append(pt.activePods[vertexName], index)
+	}
```

Comment:
Can we just track the maximum replica for every vertex? If replica 'x' succeeds, all the replicas less than 'x' should also be successful.